docs/LabVIEW_Debugging_Techniques.md

```diff
@@ -0,0 +1,95 @@
+
+![](images/ConsoleViewer1.jpg)
+
+**Figure: RT Console Viewer**
+
+The Console Viewer will show the system definition and the resulting CPU usage. The viewer is also useful for displaying debugging messages. The console viewer provides a periodic snapshot of utilization. CPU spikes and transients will probably be unobservable. If the system is very busy, it may not update the console viewer at all. You can use other debugging methods for a more accurate indication of resource utilization.
+
+As the name implies, the RT Console Viewer is only available on real-time targets. The RT Console Viewer is also available as a stand-alone add-on to LabVIEW Real-Time. See **[NI Developer Zone Tutorial: Remotely View Console Output of Real-Time Targets](https://knowledge.ni.com/KnowledgeArticleDetails?id=kA03q000000x4TjCAI&l=en-US)** for more information.
+
+&nbsp;&nbsp;&nbsp; **Note:** You cannot use the **Console Viewer** on NI Linux Real-Time targets. Instead, connect your NI Linux Real-Time targets to a computer using a serial port to view the output.
+
+#### Printing to the Console
+Printing to the console is often all that’s needed to debug an application.
+
+#### Printing With "NI VeriStand - Print Debug Line" VI
+The recommended method of printing to the console is to use "NI VeriStand - Print Debug Line" VI. This VI is part of Custom Device API » Utilities » Print Debug Line.vi.
+
+![](images/NIVSdebugString.jpg)
+
+This VI works on both Windows and RT execution hosts. It has an optional input (Attribute) to change the color of the text. It also has an optional input to append the string to the VeriStand log file.
+
+#### Printing With ni_emb.dll
+
+You’ll find **ni_emb.dll** in **&lt;labview&gt;\Targets\NI\RT\vi.lib**. This dll contains a stub function called **PrintStringToConsole**. Calling this function sends a string to the RT console. Configure the function to run in any thread using the C calling convention. The return type is void and it has a C String pointer input constant. You’ll find this function wrapped in a VI in the same folder in **rtutility.llb\RT Debug String.vi**. Since **ni_emb.dll** is a stub dll, it’s not necessary to deploy this VI to the RT target. The stub exists so the **PrintStringToConsole** function does not return an error when called on Windows.
+
+If you do not want to call **ni_emb.dll** on a Windows OS, you can use a **[Conditional Disable Structure](https://zone.ni.com/reference/en-XX/help/371361R-01/glang/conditional_disable_structure/)** around the dll. See **[NI Tutorial: Using the Conditional Disable Structure](https://zone.ni.com/reference/en-XX/help/371361R-01/lvhowto/creating_cond_disable_struc/)** for more information.
+
+![](images/Disable_ni_emb.jpg)	 
+
+**Figure: Disable ni_emb.dll for non-Windows Operating Systems**
+
```

Comment:
I would add a new section here for using the Telemetery Custom Device before mentioning the DSM tool.